salt/cloud/clouds/vmware.py

```diff
@@ -569,6 +569,34 @@ def _wait_for_task(task, vm_name, task_type, sleep_seconds=1, log_level='debug')
         raise task.info.error
 
 
+def _wait_for_host(host_ref, task_type, sleep_seconds=5, log_level='debug'):
+    time_counter = 0
+    while host_ref.runtime.connectionState != 'notResponding':
+        message = "[ {0} ] Waiting for host {1} to finish [{2} s]".format(host_ref.name, task_type, time_counter)
+        if log_level == 'info':
+            log.info(message)
+        else:
+            log.debug(message)
+        time.sleep(int(sleep_seconds))
```

Comment:
@cachedout I thought about that as well but the host system after a reboot has been initiated moves from connected state to ``notResponding`` and then has to move to  ``connected`` state back. It can take anytime greater than 4 minutes for a complete reboot. Not sure how to set a generic timeout for all people.